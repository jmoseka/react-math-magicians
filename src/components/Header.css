header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 2rem 0;
}

.magician-color {
  color: #d36a08;
}

.nav {
  list-style: none;
  display: flex;
}

.nav li {
  font-size: 1.2rem;
  margin-left: 1rem;
  font-weight: bold;
}

.line {
  border-right: 3px solid black;
  margin-left: 20px;
}
