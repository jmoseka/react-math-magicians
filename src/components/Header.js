import { Link } from 'react-router-dom';
import './Header.css';

const Header = () => (
  <header className="container">
    <h1>Math Magicians</h1>
    <ul className="nav">
      <li>
        <Link to="/home">Home</Link>
      </li>
      <span className="v-line" />
      <li><Link to="/calculator">Calculator</Link></li>
      <span className="v-line" />
      <li><Link to="/quote">Quote</Link></li>
    </ul>
  </header>
);

export default Header;
