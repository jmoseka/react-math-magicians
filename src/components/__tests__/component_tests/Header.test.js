import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../../Header';

afterEach(() => {
  cleanup();
});

test('should match snapshot', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
