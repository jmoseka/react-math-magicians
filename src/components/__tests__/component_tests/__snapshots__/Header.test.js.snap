// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<header
  className="container"
>
  <h1>
    Math
     
    <span
      className="magician-color"
    >
      Magicians
    </span>
  </h1>
  <ul
    className="nav"
  >
    <li>
      <a
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <span
      className="line"
    />
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <span
      className="line"
    />
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</header>
`;
